lib/amqp_client.js

```diff
@@ -133,21 +132,26 @@ AMQPClient.prototype.connect = function(url) {
     self._connection.on(Connection.Connected, function(c) {
       debug('connected');
       self.emit(AMQPClient.ConnectionOpened);
-      self._session = self._newSession(c);
+      if (self._session) {
+        debug('session already exists, re-using');
+        self._session.connection = self._connection;
+      } else {
+        self._session = self._newSession(c);
+        self._session.on(Session.Unmapped, function(s) {
```

Comment:
as an additional cleanup suggestion (while we're in this code here), it looks like we can move at least the Connection on('errorReceived') below to `_newConnection` as well maybe